src/main/java/io/fabric8/elasticsearch/discovery/k8s/K8sUnicastHostsProvider.java

```diff
@@ -119,67 +115,44 @@ public K8sUnicastHostsProvider(Settings settings,
     }
 
     private List<DiscoveryNode> getNodesFromKubernetesSelector(String currentIpAddress) {
-        Map<String, Pod> podMap = KubernetesHelper.getSelectedPodMap(getKubernetes(), selector);
-        Collection<Pod> pods = podMap.values();
-
-        if (pods == null) {
-            logger.trace("no pod found for selector [{}].", this.selector);
-            return cachedDiscoNodes;
-        }
-
-        for (Pod pod : pods) {
-            String status = pod.getCurrentState().getStatus();
-            logger.trace("k8s instance {} with status {} found.", pod.getId(), status);
-
-            if (!status.equals(Status.RUNNING)) {
-                logger.trace("k8s instance {} is not running - ignoring.", pod.getId());
-                continue;
-            }
-
-            try {
-                String podIp = pod.getCurrentState().getPodIP();
-
-                if (podIp.equals(currentIpAddress)) {
-                    // We found the current node.
-                    // We can ignore it in the list of DiscoveryNode
-                    logger.trace("current node found. Ignoring {} - {}", pod.getId(), podIp);
+        // get all endpoints for service
+        final Endpoints endpoints = getKubernetes().endpointsForService(this.selector, null);
```

Comment:
IAE thrown by kubernetes-api-2.1.0::KubernetesClient.validateNamespace: Template parameter set to null. 

```bash
2015-05-28T18:17:16.071899300Z [2015-05-28 18:17:16,071][WARN ][io.fabric8.elasticsearch.discovery.k8s.K8sUnicastHostsProvider] [Cayman] Exception caught during discovery java.lang.IllegalArgumentException : Template parameter value is set to null
2015-05-28T18:17:16.071899300Z java.lang.IllegalArgumentException: Template parameter value is set to null
2015-05-28T18:17:16.071899300Z 	at org.apache.cxf.jaxrs.impl.UriBuilderImpl.doBuild(UriBuilderImpl.java:97)
2015-05-28T18:17:16.071899300Z 	at org.apache.cxf.jaxrs.impl.UriBuilderImpl.buildFromEncoded(UriBuilderImpl.java:230)
2015-05-28T18:17:16.071899300Z 	at org.apache.cxf.jaxrs.client.ClientProxyImpl.invoke(ClientProxyImpl.java:180)
2015-05-28T18:17:16.071899300Z 	at com.sun.proxy.$Proxy29.endpointsForService(Unknown Source)
2015-05-28T18:17:16.071899300Z 	at io.fabric8.elasticsearch.discovery.k8s.K8sUnicastHostsProvider.getNodesFromKubernetesSelector(K8sUnicastHostsProvider.java:119)
2015-05-28T18:17:16.071899300Z 	at io.fabric8.elasticsearch.discovery.k8s.K8sUnicastHostsProvider.buildDynamicNodes(K8sUnicastHostsProvider.java:102)
2015-05-28T18:17:16.071899300Z 	at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.sendPings(UnicastZenPing.java:313)
2015-05-28T18:17:16.071899300Z 	at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing$2.doRun(UnicastZenPing.java:228)
2015-05-28T18:17:16.071899300Z 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:36)
2015-05-28T18:17:16.071899300Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
2015-05-28T18:17:16.071899300Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
2015-05-28T18:17:16.071899300Z 	at java.lang.Thread.run(Thread.java:745)
```